docs/src/main/asciidoc/config-your-quarkus-applications-using-a-yaml-file_howto.adoc

```diff
@@ -0,0 +1,407 @@
+== Overview
+
+Configuration options enable you to change the settings of your application in a single configuration file.
+
+{project-name} supports configuration profiles that you can use to group related properties and switch between profiles as required.
+
+By default, {project-name} reads properties from the `application.properties` file located in the `src/main/resources` directory.
+Instead, you can configure {project-name} to read properties from a YAML file.
+
+When you add the `quarkus-config-yaml` dependency to your project `pom.xml` file, you can configure and manage your application properties in the `application.yaml` file.
+
+For more information, see the xref:proc_enabling_yaml_configuration[Enabling YAML configuration].
+
+{project-name} supports YAML configuration files through the `SmallRye Config` implementation of Eclipse MicroProfile Config.
+
+You can use the link:https://microprofile.io/project/eclipse/microprofile-config[MicroProfile Config] specification from the Eclipse MicroProfile project to inject configuration properties into your application and access them using a method defined in your code.
+
+{project-name} can also read application properties from different origins, including the following sources:
+
+* The file system
+* A database
+* A Kubernetes or OpenShift `ConfigMap` or Secret object
+* Any source that a Java application can load
+
+// WAS include::modules/proc_adding-yaml-configuration-support.adoc[leveloffset=+1]
+// include::modules/proc_enabling_yaml_configuration.adoc[leveloffset=+1]
+[id="proc_enabling_yaml_configuration"]
+== Enabling YAML configuration
+
+To enable YAML configuration, add the `quarkus-config-yaml` extension:
+
+:add-extension-extensions: quarkus-config-yaml
+include::{includes}/devtools/extension-add.adoc[]
+
+You can also just add the following dependency into your project:
+
+[source,xml,role="primary asciidoc-tabs-target-sync-cli asciidoc-tabs-target-sync-maven"]
+.pom.xml
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-config-yaml</artifactId>
+</dependency>
+----
+
+[source,gradle,role="secondary asciidoc-tabs-target-sync-gradle"]
+.build.gradle
+----
+implementation("io.quarkus:quarkus-config-yaml")
+----
+
+Remove the `src/main/resources/application.properties` and create a `src/main/resources/application.yaml` file.
+
+NOTE: If both are present, Quarkus prioritizes configuration properties from the YAML file first and then from the
+Properties file. However, to avoid confusion, we recommend removing the Properties file.
+
+TIP: Quarkus supports both the  `yml` and `yaml` file extensions.
+
+// include::modules/proc_using-nested-object-configuration-with-yaml.adoc[leveloffset=+2]
+
+[id="proc_using-nested-object-configuration-with-yaml"]
+
+== Using nested object configuration with YAML
+
+In a YAML configuration file, you can define a nested class inside an existing class to set configuration properties for your {project-name} application.
```

Comment:
I'm not sure `class` is the right word here? But I'm not very familiar with YAML terminology.

Reply:
Probably nested objects / elements.

Reply:
Updated it to element. Thanks!